src/Stratis.Bitcoin.Features.Wallet/WalletManager.cs

```diff
@@ -536,20 +536,52 @@ public IEnumerable<AccountBalance> GetBalances(string walletName, string account
                 
                 foreach (var account in accounts)
                 {
-                    (Money AmountConfirmed, Money AmountUnconfirmed) result = account.GetSpendableAmount();
+                    (Money amountConfirmed, Money amountUnconfirmed) result = account.GetSpendableAmount();
 
                     balances.Add(new AccountBalance
                     {
                         Account = account,
-                        AmountConfirmed = result.AmountConfirmed,
-                        AmountUnconfirmed = result.AmountUnconfirmed
+                        AmountConfirmed = result.amountConfirmed,
+                        AmountUnconfirmed = result.amountUnconfirmed
                     });
                 }
             }
 
             return balances;
         }
 
+        /// <inheritdoc />
+        public AddressBalance GetAddressBalance(string address)
+        {
+            Guard.NotEmpty(address, nameof(address));
+            this.logger.LogTrace("({0}:'{1}')", nameof(address), address);
+
+            AddressBalance balance = new AddressBalance
+            {
+                Address = address,
+                CoinType = this.coinType
+            };
+
+            lock (this.lockObject)
+            {
+                foreach (Wallet wallet in this.Wallets)
+                {
+                    if (!wallet.ContainsAddress(new HdAddress { Address = address })) continue;
+
+                    HdAddress hdAddress = wallet.GetAllAddressesByCoinType(this.coinType).FirstOrDefault(a => a.Address == address);
```

Comment:
I suggest you remove this line: `if (!wallet.ContainsAddress(new HdAddress { Address = address })) continue;`
and do the check after `HdAddress hdAddress = wallet.GetAllAddressesByCoinType(this.coinType).FirstOrDefault(a => a.Address == address);`, like in my previous comment. this way you only search for the address once.

Reply:
I agree, but then there is a possibility that hdAddress is in fact NULL, correct?